src/codegate/providers/ollama/completion_handler.py

```diff
@@ -82,17 +82,26 @@ async def ollama_stream_generator(  # noqa: C901
 
 class OllamaShim(BaseCompletionHandler):
 
-    def __init__(self, base_url):
-        self.client = AsyncClient(host=base_url, timeout=300)
-
     async def execute_completion(
         self,
         request: ChatCompletionRequest,
+        base_url: Optional[str],
         api_key: Optional[str],
         stream: bool = False,
         is_fim_request: bool = False,
     ) -> Union[ChatResponse, GenerateResponse]:
         """Stream response directly from Ollama API."""
+
+        if not base_url:
+            raise ValueError("base_url is required for Ollama")
```

Comment:
could we default to `localhost:11434` ?

Reply:
We should already get this default frm the caller.